src/zarr/codecs/sharding.py

```diff
@@ -98,6 +98,9 @@ async def set(self, value: Buffer, byte_range: ByteRangeRequest | None = None) -
     async def delete(self) -> None:
         del self.shard_dict[self.chunk_coords]
 
+    async def setdefault(self, default: Buffer) -> None:
+        self.shard_dict.setdefault(self.chunk_coords, default)
```

Comment:
I don't think this is actually tested anywhere. I'm not 100% sure, but I think all the Group / Array Metadata creation method will be using `Store` as their `ByteSetter`.